src/main/java/com/game/Battleship/repository/BoardRepository.java

```diff
@@ -0,0 +1,66 @@
+package com.game.Battleship.repository;
+import com.game.Battleship.models.*;
+import org.springframework.stereotype.Service;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@Service
+public class BoardRepository {
+
+    private final PlayerBoard playerBoard;
+    private final ComputerBoard computerBoard;
+    private final List<Board> boardList;
+
+    public BoardRepository(){
+        this.playerBoard = new PlayerBoard();
+        this.computerBoard = new ComputerBoard();
+        this.boardList = new ArrayList<>(Arrays.asList(
+                this.playerBoard,
+                this.computerBoard
+        ));
+
+    }
+
+    public List<Board> allBoards(){
+        return boardList;
+    }
+
+    public Board getPlayerBoard() { return this.playerBoard; }
+
+    public Board getComputerBoard() { return this.computerBoard;}
+
+    public Board updateBoardWithShips(Board board, Ship ship, Integer row, Integer column){
+
+        Integer index = convertCoordinatesToIndex(board, row, column);
+
+        board.addShip(ship, index);
+        return board;
+
+    }
+
+    public Board updateBoardWithStrike(Board board, Integer row, Integer column){
+
+            Integer index = convertCoordinatesToIndex(board, row, column);
+
+            Cell cell = board.getBoardField().get(index);
+
+            if (cell.getValue() == null ) {
+                cell.setValue("Missed");
+            } else if ("Ship".equals(cell.getValue()) ) {
+                cell.setValue("Hit");
+            }
+
+            board.getBoardField().set(index, cell);
+            return board;
+
+    }
+
+    public Integer convertCoordinatesToIndex(Board board, Integer row, Integer column){
+        Integer index = row * board.getSize() + column;
+        return index;
+    }
+
+
```

Comment:
only need one line at the end of a file.

Reply:
Done